docs/doc/21-load-data/index.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Load Data into Databend
```

Comment:
How about in this way?

We have two loading methods: `Push` and `Pull`.

## Push

Users push data into the databend actively.

- Streaming Load API
- Upload to stage API
- Presign and upload, then call COPY FROM user stage or internal stage

## Pull

Databend pulls data from users' storage.

- Create an external stage and COPT FROM it.
- COPY FROM an external location

Supported locations include:

- local fs
- s3 (aws s3, minio, ...)
- gcs
- http
- ipfs
- ...

Reply:
That's quite of a new way of look at data loading! Push and Pull sound a little bit more API words that might be confusing. But I think we can do that in a blog, setting a point of view about data loading.

Reply:
If so, we need to find a way of representing `Supported File Locations`. In current way, they could be more confusing.

- `Stage`: the stage is not a location. It's a reference to location. For the user stage and internal stage, it always refers to the exact location the databend-query use; For the external stage, it refers to the location that the user specified in `CREATE STAGE`.
- `Bucket or container`, `Local file system`, `Remote server`: they are the same thing.

Reply:
Can you share your thoughts how to exactly document the Supported File Locations?

Reply:
ok, then how would users get to know where they can load from ?

Reply:
> ok, then how would users get to know where they can load from ?

Nice question.

This why I want to build the loading method ideas about `push` and `pull`. Users need to decide which method they want to use first.

- For `push`, users have their data in MySQL or application, they want to push to databend query. In this way, users will handle the location of data.
- For `pull`, users will store data in somewhere and let databend query to pull. And we can support A, B, and C locations.

Without the loading methods idea, users don't have the context of file location.

Reply:
> 

I prefer this version you suggested, as the PUSH and Pull one seems a little complicated

Reply:
Supported File Locations
Databend can load data from files that are stored in local file system, [Supported Object Storage Solutions](https://file+.vscode-resource.vscode-cdn.net/Users/eric/Documents/GitHub/rsdoc/docs/doc/10-deploy/00-understanding-deployment-modes.md#supported-object-storage-solutions), and remote servers.